service/raft/join_node.hh

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <set>
+#include <seastar/core/sstring.hh>
+#include "raft/raft.hh"
+
+namespace service {
+
+struct join_node_request_params {
+    raft::server_id host_id;
+    std::optional<raft::server_id> replaced_id;
+    std::vector<sstring> ignore_nodes;
+    sstring cluster_name;
+    sstring snitch_name;
+    sstring datacenter;
+    sstring rack;
+    sstring release_version;
+    uint32_t num_tokens;
+    uint32_t shard_count;
+    uint32_t ignore_msb;
+    std::vector<sstring> supported_features;
+};
+
+struct join_node_request_result {
+    // Request was successfully placed and will be processed
+    // by the topology coordinator.
+    struct ok {};
+
+    // The request was already accepted by the topology coordinator.
+    struct accepted {};
```

Comment:
Looks like my guess was exactly wrong, so "accepted" should be "already_a_member" to reduce ambiguity.

Reply:
Now there is no `accepted` response, only `ok` + `rejected`, so I guess it's not that ambiguous anymore.